src/resmom_win/start_exec.c

```diff
@@ -1768,7 +1767,7 @@ finish_exec(job *pjob)
 	if (pjob->ji_user->pw_userlogin != INVALID_HANDLE_VALUE) {
 		if (!impersonate_user(pjob->ji_user->pw_userlogin)) {
 			sprintf(log_buffer,
-				"failed to ImpersonateLoggedOnUser on %s", mom_host);
+				"failed to ImpersonateLoggedOnUser on %s, error=%ld", mom_host, GetLastError());
```

Comment:
NOTE to reviewers: This change will be reflected in the comment field of held windows jobs. I wanted to capture logon failure codes in the comment itself, but looks like the the logon failure codes appear in logon_pw and the pw_userlogin will be null. so in reality I'm not getting the correct code in the comment, but just the code 87 for ERROR_INVALID_PARAMETER which I'm setting explicitly in impersonate_user as part of this PR.

```
Job ID                         Username        Queue           Jobname         SessID   NDS  TSK   Memory Time  S Time
------------------------------ --------------- --------------- --------------- -------- ---- ----- ------ ----- - -----
7.server                         pbsuser         workq           STDIN                --     1     1    --    --  H   --
   failed to ImpersonateLoggedOnUser on winmom1, error=87:job has bad password
```

So Thinking of reverting this line. please give your opinions

Reply:
In my opinion you can modify the if check and add "pjob->ji_user->pw_userlogin != NULL" and before the else block can verify for the same and log accordingly. And also keep the current else block to cover the situation when we get a valid handle but Windows API failed.
Also can keep this block's error login message as it was previously.

Reply:
fixed in https://github.com/openpbs/openpbs/pull/1794/commits/5ca363a538c69345193a7ce576461e4932416e0b
Please +1 if you agree to the fix

Reply:
Do you think it will be better, if we add an "else if block" after this to verify whether the handle is NULL and log accordingly ?

Reply:
I don't think its better, because both `NULL` and `INVALID_HANDLE_VALUE` are non valid values and we never differentiate them from each other while logging in many other places, I just wish the original code was more confident to check for just either `NULL` or `INVALID_HANDLE_VALUE`, so I doubt it will add any more helpful info. over that the code will look ugly. +1 if you agree

Reply:
If this additional error code is not the right one, or even adding any more value, then why add this now - we should revert back to what it was

Reply:
reverted it and added an error log instead to log the error code. https://github.com/openpbs/openpbs/pull/1794/commits/aa46739ce40d19359d847bc5530e632a383fa4c4